### ISSUE TITLE
release: 19.2.8

### ISSUE THREAD

**Issue by asubiotto** [cockroachdb/cockroach#50465]
Candidate SHA: 04216787bfef5c0a577d93b16c9e91fd44637ecf
Deployment status: Qualifying
Qualification Suite: https://teamcity.cockroachdb.com/viewType.html?buildTypeId=Cockroach_ReleaseQualification&tab=buildTypeStatusDiv&branch_Cockroach=provisional_202006230817_v19.2.8
Nightly Suite: https://teamcity.cockroachdb.com/viewType.html?buildTypeId=Cockroach_Nightlies_NightlySuite&tab=buildTypeStatusDiv&branch_Cockroach_Nightlies=provisional_202006230817_v19.2.8
- Roachtest GCE failed due to quota issues. Running independently: https://teamcity.cockroachdb.com/viewLog.html?buildId=2032238&buildTypeId=Cockroach_Nightlies_WorkloadNightly
 
Admin UI for Qualification Clusters:
- Old Version: http://alfonso-release-v1927-2e19ff0-0001.roachprod.crdb.io:26258/#/metrics/overview/cluster
- New Version: http://alfonso-release-v1928-0421678-0001.roachprod.crdb.io:26258/#/metrics/overview/cluster
 
[Release process checklist](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process)
 
Prep date: `Monday 6/22/2020`
- [x] [Pick a SHA](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-PickingaSHA)
  - [x] fill in `Candidate SHA` above
  - [x] email thread on releases@
- [x] [Tag the provisional SHA](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-TagtheprovisionalSHA)
- [x] [Publish provisional binaries](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-Publishtheprovisionalbinaries)
- [x] Ack security@ on the generated Stackdriver Alert to confirm these writes were part of a planned release (Just reply on the email received alert email acking that this was part of the release process)
 
Release Qualification
- [x] [Check binaries](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-Checkbinaries)
    - https://github.com/cockroachdb/bincheck/pull/202
- [x] [Deploy to test clusters](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-DeploytoTestClusters)
- [x] [Verify node crash reports](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-Verifynodecrashreportsappearinsentry.io)
    - https://sentry.io/organizations/cockroach-labs/issues/1258371060/?project=164528&query=panic&statsPeriod=14d
- [x] [Start release qualification suite](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-RuntheSuiteofNightlyRoachtests)
- [x] [Start nightly suite](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-RuntheSuiteofNightlyRoachtests)
- [x] fill in all WIP elements above (Nightly Suite, Admin UI, etc.)
 
One day after prep date:
- [x] [Get signoff on roachtest failures](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-RoachtestFailureSignoff)
 
- [x] [Keep an eye on clusters](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/328859690/Release+Qualification#Monitoring-the-Qualification-Clusters) until release date. Do not proceed below until the release date.
 
Release date: `Monday 6/29/2020`
- [x] [Check cluster status](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/328859690/Release+Qualification#Monitoring-the-Qualification-Clusters)
- [x] [Tag release](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-Tagtherelease)
- [x] [Bless provisional binaries](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-Blesstheprovisionalbinaries)
- [x] If applicable, [update the map that roachtests use to map a version to a previous version, to reference the newly tagged version](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/test_runner.go#L1145)
    - https://github.com/cockroachdb/cockroach/pull/50770
 
- For production or stable releases in the latest major release series
  - ~Update the [Homebrew Formula](https://github.com/cockroachdb/homebrew-tap/blob/master/Formula/cockroach.rb)~
  - ~Update the image tag in our [Orchestrator configurations](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-Orchestratorconfigurations)~
 
- For production or stable releases
  - [x] Announce version to the [registration cluster](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-AnnounceVersionToRegCluster)
 
- [x] Update docs
    - https://www.cockroachlabs.com/docs/releases/v19.2.8.html
- [x] External communications for release
- [x] Clean up provisional tag from repository

**Comment by asubiotto** [comment]
Restarting with a SHA that includes the security fix: 04216787bfef5c0a577d93b16c9e91fd44637ecf

**Comment by asubiotto** [comment]
Looks like Roachtest GCE nightly could not be started due to exceeding a CPU quota. Can we restart just that part of the suite? (cc @jlinder) We also have the option to pass different roachprod zones to the build, but I'm slightly confused since I only see the option to do so at the Nightly Suite level, not the Roachtest GCE level (although I haven't looked very hard).

**Comment by asubiotto** [comment]
Looks like it still ran into an issue, we're running close to the 4k cpus limit in us-central. Will move zones around and see what happens.

**Comment by asubiotto** [comment]
Starting the test failure checkoff process while the build is still running to save time.

# Test Failures List

## Roachtest GCE

Failures: https://teamcity.cockroachdb.com/viewLog.html?buildId=2032238&buildTypeId=Cockroach_Nightlies_WorkloadNightly

### [kv]

- [ ] kv/contention/nodes=4

**Comment by tbg** [comment]
> 	ts_util.go:130,kv.go:259,cluster.go:2460,errgroup.go:57: spent 47.368421% of time below target of 100.000000 txn/s, wanted no more than 5.000000%

@nvanbenschoten looks like your wheelhouse

**Comment by rafiss** [comment]
Signed off on Django.

**Comment by jlinder** [comment]
I suspect the quota issue was hit because we are running three releases at the same time on top of the normal nightlies. I've put in a request to raise the quota on CPU as well as in-use IPs and local SSDs as those numbers were close enough that the limit might be reached when running all three as well.

**Comment by jlinder** [comment]
The quota limit increases were approved.

**Comment by nvanbenschoten** [comment]
`kv/contention/nodes=4` has always been flaky on `v19.2`. See https://github.com/cockroachdb/cockroach/issues/40786 and https://teamcity.cockroachdb.com/project.html?projectId=Cockroach_Nightlies&buildTypeId=&tab=testDetails&testNameId=-9215103075698950051&order=START_DATE_DESC&branch_Cockroach_Nightlies=release-19.2&itemsCount=100. Last time @irfansharif looked, he diagnosed that it had something to do with fairness issues around the `contentionQueue`. This was part of the reason we redesigned this in v20.1.

I think we should reduce the aggressiveness of the test to avoid some of the starvation that results from these fairness issues under such severe contention. In the meantime, signing off.

**Comment by tbg** [comment]
tpccbench timed out. It did run some of the workloads, but look at the last one's logs:

```
Initializing 2100 connections...
Initializing 10500 workers and preparing statements...
I200623 22:47:28.365328 1 workload/cli/run.go:362  retrying after error while creating load: preparing 
		UPDATE district
		SET d_next_o_id = d_next_o_id + 1
		WHERE d_w_id = $1 AND d_id = $2
		RETURNING d_tax, d_next_o_id: EOF
Initializing 2100 connections...
Initializing 10500 workers and preparing statements...
I200623 22:48:30.563745 1 workload/cli/run.go:362  retrying after error while creating load: preparing 
		UPDATE district
		SET d_next_o_id = d_next_o_id + 1
		WHERE d_w_id = $1 AND d_id = $2
		RETURNING d_tax, d_next_o_id: EOF
Initializing 2100 connections...
I200623 22:49:34.501519 1 workload/cli/run.go:362  retrying after error while creating load: EOF
Initializing 2100 connections...
Initializing 10500 workers and preparing statements...
```

This basically just goes on and on, we never manage to prep all statements before a node gets chaos-killed. @nvanbenschoten is that something we're aware of? I feel like I've seen this test time out in that way a few times. What do we do to fix? Is the chaos timing too aggressive? I assume it's not possible (feasible) to prepare the statements before starting the workload+chaos.

**Comment by ajwerner** [comment]
The schema change one seems potentially bad:

`schemachange.go:476,schemachange.go:439,cluster.go:2460,errgroup.go:57: pq: foreign key violation: "district" row d_w_id=288, d_id=1 has no match in "warehouse"`

I cannot imagine how that's true.

**Comment by ajwerner** [comment]
Chalking the schema change failure up as https://github.com/cockroachdb/cockroach/issues/44301 which has been newly prioritized.

**Comment by nvanbenschoten** [comment]
> This basically just goes on and on, we never manage to prep all statements before a node gets chaos-killed. @nvanbenschoten is that something we're aware of? I feel like I've seen this test time out in that way a few times. What do we do to fix? Is the chaos timing too aggressive? I assume it's not possible (feasible) to prepare the statements before starting the workload+chaos.

I think there's something more going on. We've dropped the chaos aggressiveness in the past and it didn't seem to help. These prepared statements are usually very quick, so it's surprising to see them stall for over a minute. It makes me wonder if they're getting stuck in some backoff loop if they start the process at the wrong time.

I'll confirm the part about prepared statements being quick though. We are initializing 2100 connections and 10500 workers, so maybe I'm misremembering how long that scale takes. There was also some movement in this area when we moved tpcc to pgx. That could be related.
